controllers/activemqartemis_controller2_test.go

```diff
@@ -0,0 +1,319 @@
+				content, err := RunCommandInPod(podWithOrdinal, brokerCr.Name+"-container", newFileCmd)
+				Expect(err).To(BeNil())
+				Expect(*content).To(BeEmpty())
+
+				By("check the file exists")
+				lsCmd := []string{"ls", "/amq/extra/pvcs/mydata"}
+				content, err = RunCommandInPod(podWithOrdinal, brokerCr.Name+"-container", lsCmd)
+				Expect(err).To(BeNil())
+				Expect(*content).To(ContainSubstring("fake.config"))
+
+				By("shut down the pod")
+				CleanResource(createdBrokerCr, createdBrokerCr.Name, defaultNamespace)
+
+				By("deploying 2 brokers to use existing volume")
+				brokerCr, createdBrokerCr = DeployCustomBroker(defaultNamespace, func(candidate *brokerv1beta1.ActiveMQArtemis) {
+					candidate.Spec.DeploymentPlan.Size = common.Int32ToPtr(2)
+					candidate.Spec.DeploymentPlan.ReadinessProbe = &corev1.Probe{
+						InitialDelaySeconds: 1,
+						PeriodSeconds:       1,
+						TimeoutSeconds:      5,
+					}
+					candidate.Spec.DeploymentPlan.ExtraVolumes = []corev1.Volume{
+						{
+							Name: "mydata",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: "my-pvc-0",
+								},
+							},
+						},
+					}
+					candidate.Spec.DeploymentPlan.ExtraVolumeMounts = []*corev1.VolumeMount{
+						{
+							Name:      "mydata",
+							MountPath: "/opt/common",
+						},
+					}
+				})
+
+				By("waiting 2 pods ready")
+				WaitForPod(brokerCr.Name, 0, 1)
+
+				By("checking the file still exist on volumes")
+				lsCmd = []string{"ls", "/opt/common"}
+				catCmd := []string{"cat", "/opt/common/fake.config"}
+				for podOrdinal := range []int32{0, 1} {
+					podWithOrdinal = namer.CrToSS(brokerCr.Name) + "-" + strconv.Itoa(podOrdinal)
+					content, err = RunCommandInPod(podWithOrdinal, brokerCr.Name+"-container", lsCmd)
+					Expect(err).To(BeNil())
+					Expect(*content).To(ContainSubstring("fake.config"), *content)
+
+					content, err = RunCommandInPod(podWithOrdinal, brokerCr.Name+"-container", catCmd)
+					Expect(err).To(BeNil())
+					Expect(*content).To(ContainSubstring("Hello-World"))
+				}
+
+				CleanResource(createdBrokerCr, createdBrokerCr.Name, defaultNamespace)
+				CleanResource(createdPvc, pvc.Name, pvc.Namespace)
+			}
+		})
+
+		It("external pvc with pvc templates", func() {
+			if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+
+				brokerCrName := NextSpecResourceName()
+
+				pvcName := "mydata-" + namer.CrToSS(brokerCrName) + "-0"
+				By("create extra pvc: " + pvcName)
+				pvc, createdPvc := DeployCustomPVC(pvcName, defaultNamespace, func(candidate *corev1.PersistentVolumeClaim) {
+
+					candidate.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					}
+					candidate.Spec.Resources = corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("2Mi"),
+						},
+					}
+				})
```

Comment:
is this PVC used?

Reply:
yes it's used in test. "my-data" maps to the base name of the pvc template and is mounted for test to write some bytes and verify.